### PR TITLE
Support K8s Driver/Executor Cores Request Conf

### PR DESCRIPTION
**Pull request by akhurana001**
# TL;DR
Use Spark k8s request conf to determine cores limit for driver/executors: https://github.com/lyft/flyte/issues/632

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/632

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: go/tasks/plugins/k8s/spark/spark.go [hunk 1]
```diff
@@ -137,12 +137,24 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
 	}
 
 	// Set pod limits.
-	if sparkConfig["spark.kubernetes.driver.limit.cores"] == "" && sparkConfig["spark.driver.cores"] != "" {
-		sparkConfig["spark.kubernetes.driver.limit.cores"] = sparkConfig["spark.driver.cores"]
+	if sparkConfig["spark.kubernetes.driver.limit.cores"] == "" {
```
Comment (EngHabu) [inline go/tasks/plugins/k8s/spark/spark.go:140]: nit:
```suggestion
	if len(sparkConfig["spark.kubernetes.driver.limit.cores"]) == 0 {
```

File: go/tasks/plugins/k8s/spark/spark.go [hunk 2]
```diff
@@ -137,12 +137,24 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
...
+		// spark.kubernetes.driver.request.cores takes precedence over spark.driver.cores
+		if sparkConfig["spark.kubernetes.driver.request.cores"] != "" {
+			sparkConfig["spark.kubernetes.driver.limit.cores"] = sparkConfig["spark.kubernetes.driver.request.cores"]
+		} else if sparkConfig["spark.driver.cores"] != "" {
+			sparkConfig["spark.kubernetes.driver.limit.cores"] = sparkConfig["spark.driver.cores"]
```
Comment (EngHabu) [inline go/tasks/plugins/k8s/spark/spark.go:145]: Should you set the requests to match this same number?

Comment (akhurana001) [reply go/tasks/plugins/k8s/spark/spark.go:145]: Don't think that's necessary as It's ok to have either `spark.kubernetes.driver.request.cores` or `spark.driver.cores` in the conf i.e. one is good enough. 
If both are present, `spark.kubernetes.driver.request.cores` has precedence over `spark.driver.cores`.

Comment (EngHabu) [reply go/tasks/plugins/k8s/spark/spark.go:145]: SG

File: go/tasks/plugins/k8s/spark/spark.go (hunk 1 above)
Comment (akhurana001) [reply go/tasks/plugins/k8s/spark/spark.go]: replaced the check everywhere with len()

**Review by EngHabu — APPROVED** [review]

**Pull request merged**